pkg/devfile/devfile.go

```diff
@@ -734,3 +741,132 @@ func UpdateLocalDockerfileURItoAbsolute(devfile data.DevfileData, dockerfileURL
+				log.Error(err, "failed to validate devfile")
+				return shouldIgnoreDevfile, nil, err
+			}
+		}
+		// TODO: if only one kube component, should return a warning that no deploy command being defined
+	}
+	imageCompFilter := common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{
+			ComponentType: v1alpha2.ImageComponentType,
+		},
+	}
+	imageComp, err := devfileData.GetComponents(imageCompFilter)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to get image component from %s", URL))
+		return shouldIgnoreDevfile, nil, fmt.Errorf(fmt.Sprintf("err: %v, failed to get image component from %s", err, URL))
+	}
+	if len(imageComp) == 0 {
+		log.Info(fmt.Sprintf("Found 0 image components being defined in devfile from %s, it is not a valid outerloop definition, the devfile will be ignored. A devfile will be matched from registry...", URL))
+		shouldIgnoreDevfile = true
+		return shouldIgnoreDevfile, nil, nil
+	} else {
+		if len(imageComp) > 1 {
+			found := false
+			for _, component := range imageComp {
+				if component.Image != nil && component.Image.Dockerfile != nil && component.Image.Dockerfile.DockerfileSrc.Uri != "" {
+					dockerfileURI := component.Image.Dockerfile.DockerfileSrc.Uri
+					absoluteURI := strings.HasPrefix(dockerfileURI, "http://") || strings.HasPrefix(dockerfileURI, "https://")
+					if absoluteURI {
+						// image uri
+						_, err = util.CurlEndpoint(dockerfileURI)
+					} else {
+						if devfileSrc.Path != "" {
+							// local devfile src with relative dockerfile uri
+							dockerfileURI = path.Join(path.Dir(URL), dockerfileURI)
+							err = parserUtil.ValidateFile(dockerfileURI)
+						} else {
+							// remote devfile src with relative dockerfile uri
+							var u *url.URL
+							u, err = url.Parse(URL)
+							if err != nil {
+								log.Error(err, fmt.Sprintf("failed to parse from %s", URL))
+								return shouldIgnoreDevfile, nil, fmt.Errorf(fmt.Sprintf("failed to parse from %s", URL))
+							}
+							u.Path = path.Join(u.Path, dockerfileURI)
+							dockerfileURI = u.String()
+							_, err = util.CurlEndpoint(dockerfileURI)
+						}
+					}
+					if err != nil {
+						log.Error(err, fmt.Sprintf("failed to curl from the dockerfile URI %s, invalid image component: %s", URL, component.Name))
+						return shouldIgnoreDevfile, nil, fmt.Errorf(fmt.Sprintf("failed to get dockerfile from the URI %s, invalid image component: %s", URL, component.Name))
+					}
+				}
+				if _, ok := deployCompMap[component.Name]; ok {
+					found = true
+					break
+				}
+			}
+			if !found {
+				err = fmt.Errorf("found more than one image components, but no deploy command associated with any being defined in the devfile from %s", URL)
+				log.Error(err, "failed to validate devfile")
+				return shouldIgnoreDevfile, nil, err
+			}
+		}
+		// TODO: if only one image component, should return a warning that no apply command being defined
```

Comment:
Does this still need to be done?

Reply:
same as above, we currently have no way to return warning to HAC